### DISCUSSION TITLE
UseTransitions with useQueryStates()

### DISCUSSION THREAD

**Discussion by Jbenzs** [47ng/nuqs#509] [Q&A] [answered]
Does the useTransition work with useQueryStates? I test it with useQueryState and it worked as expected in the docs but if i set it like this 

```ts
const [pagination,setPagination] = useQueryStates({page: parseAsInteger.withOptions({startTransition}), pageSize: parseAsInteger.withOptions({startTransition})})
```

The transition isLoading doesnt seem to trigger in the case of query keys that should always move together.

If I am using it wrong, that is highly likely. If anyone knows that was be awesome!

**Comment by franky47** [comment, marked as answer]
In `useQueryStates`, the options are passed not per-parser, but at the global level in a second parameter, so this should work:

```ts
const [pagination, setPagination] = useQueryStates(
  {
    page: parseAsInteger,
    pageSize: parseAsInteger
  },
  {
    // Options go here
    startTransition
  }
)
```

It's a bit confusing though, since there would be three places to define options (per parser with `.withOptions`, in the second parameter to apply to all keys, and at the call level). Not sure what the proper override order should be between the first two. Probably (in order of precedence) call level, then parser level, then hook level, then defaults.

Most of the time, setting an option will apply to the whole update anyway (like `history: 'push'` or `shallow: false`).

Anyway, this calls for a docs update and fixing the parser-level options not being accounted for, thanks for fishing this one out!

**Comment by Jbenzs** [comment]
This worked! thanks a lot for that clarification :D
